src/sagemaker/estimator.py

```diff
@@ -898,18 +906,20 @@ def transformer(
                 If not specified, this setting is taken from the estimator's
                 current configuration.
             model_name (str): Name to use for creating an Amazon SageMaker
-                model. If not specified, the name of the training job is used.
+                model. If not specified, the estimator generates a default job name
+                based on the training image name and current timestamp.
         """
         tags = tags or self.tags
 
+        self._ensure_base_job_name()
+        model_name = model_name or name_from_base(self.base_job_name)
```

Comment:
I've seen this logic elsewhere. Is it worth defining a method to capture and re-use?